lib/private/AppFramework/Http/Dispatcher.php

```diff
@@ -90,9 +90,11 @@ public function __construct(Http $protocol,
 	 * @param Controller $controller the controller which will be called
 	 * @param string $methodName the method name which will be called on
 	 *                           the controller
-	 * @return array $array[0] contains a string with the http main header,
-	 *               $array[1] contains headers in the form: $key => value, $array[2] contains
-	 *               the response output
+	 * @return array $array[0] contains the http status header as a string,
+	 *               $array[1] contains response headers as an array, 
+	 *               $array[2] contains response cookies as an array,
+  	 *               $array[3] contains the response output as a string,
+	 *               $array[4] contains the response object
```

Comment:
```suggestion
	 * @return array $array[0] contains the http status header as a string,
	 *               $array[1] contains response headers as an array, 
	 *               $array[2] contains response cookies as an array,
  	 *               $array[3] contains the response output as a string,
	 *               $array[4] contains the response object
	 * @psalm-return list{0: string, 1: array<string, mixed>, 2: array<string, array<string, mixed>>, 3: string, 4: Response}
```
You could also add this typing if you're already at it.